### PR TITLE
feat: add more properties with the typed API in Style

### PR DESCRIPTION
**Pull request by tltv**
Fixes: #17729

### CODE REVIEW

**Comment by tltv** [comment]
Set to draft for now to think about if it makes sense 
- to use `ElementConstants` in the `Style`
- to add getters (there's none other than `get(String)` at the moment)

**Comment by tltv** [comment]
> - to add getters (there's none other than `get(String)` at the moment)

Dropping this idea after discussing with team. Style is only server side state that writes one way to the client. Adding all property getters could give an wrong impression of two-way synchronization.

**Comment by mstahv** [comment]
I pushed in some "dogfooding" of the new API + one overload for setFontWeight.

I don't like the usage of separate constants, makes the code less readable and I don't believe it will help in maintenance either. But this really doesn't affect the end user (nor JVM), so reason to overthink this. The PR looks perfect to me 💪

File: flow-tests/vaadin-spring-tests/test-spring-security-flow/src/main/java/com/vaadin/flow/spring/flowsecurity/views/PublicView.java [hunk 1]
```diff
@@ -26,7 +26,7 @@ public PublicView() {
         header.getStyle().set("text-align", "center");
         add(header);
         Image image = new Image("public/images/bank.jpg", "Bank");
-        image.getStyle().set("max-width", "100%").set("min-height", "0");
+        image.getStyle().setMaxWidth("100%").setMinWidth("0");
```
Comment (tltv) [inline flow-tests/vaadin-spring-tests/test-spring-security-flow/src/main/java/com/vaadin/flow/spring/flowsecurity/views/PublicView.java:29]: Did you change `min-height` to `min-width` in purpose here? Does not seem to affect test results though.

**Review by tepi — APPROVED** [review]

**Pull request merged**